packages/prettier-plugin-liquid/src/test/liquid-tag-capture-whitespace/index.liquid

```diff
@@ -0,0 +1,37 @@
+It should never break on capture var name
+printWidth: 1
+{% capture varName %}{% endcapture %}
+{%   capture   varName   %}{% endcapture %}
+{%capture varName %}{% endcapture %}
+{%- capture
+varName
+-%}{% endcapture %}
+
+It should be indentation sensitive
+<p>{% capture var %}
+    # heading
+
+    - list
+      * elem
+  {% endcapture %}
+</p>
+
+It should behave like a pre tag if the capture tag is set white-space pre
+printWidth: 1
+<!-- white-space: pre -->
+{% capture var %}
+      <div>{{ x }}</div>
+{% endcapture %}
+
+It should behave like a pre tag if the capture tag is set white-space pre (Liquid)
+printWidth: 1
+{% # white-space: pre %}
+{% capture var %}
+      <div>{{ x }}</div>
+{% endcapture %}
+
+It should print as is without the white-space pre comment
+printWidth: 1
+{% capture var %}
+      <div>{{ x }}</div>
+{% endcapture %}
```

Comment:
There should be a test with the `captureWhitespace` option, you can do that like this:
```
It should bla bla bla
captureWhitespace: 'ignore'
...
```

To verify that the code works as expected.

Reply:
my bad should be a JSON parseable value on the right hand side 
```
captureWhitespace: "ignore"
```

Reply:
Hi @charlespwd,

Do you mean like this?

```liquid
It should never break on capture var name
captureWhitespace: "ignore"
{% capture varName %}{% endcapture %}
{%   capture   varName   %}{% endcapture %}
{%capture varName %}{% endcapture %}
{%- capture
varName
-%}{% endcapture %}

It should be indentation sensitive
<p>{% capture var %}
    # heading

    - list
      * elem
  {% endcapture %}
</p>

It should behave like a pre tag if the capture tag is set white-space pre
captureWhitespace: "ignore"
<!-- white-space: pre -->
{% capture var %}
      <div>{{ x }}</div>
{% endcapture %}

It should behave like a pre tag if the capture tag is set white-space pre (Liquid)
captureWhitespace: "ignore"
{% # white-space: pre %}
{% capture var %}
      <div>{{ x }}</div>
{% endcapture %}

It should print as is without the white-space pre comment
captureWhitespace: "ignore"
{% capture var %}
      <div>{{ x }}</div>
{% endcapture %}

```

Sorry if it's a stupid question. This is actually the first time I'm ever contributing. Also have to get used a little bit on how it works.

Reply:
> Sorry if it's a stupid question.

It absolutely isn't! I'm impressed you managed to pick it up without more guidance. 

So the way those tests are setup is a bit weird because we want to test before/after. So 
- the index.liquid file is the before
- the fixed.liquid file is the after

Each tests are setup in "groups" which are the following:
- A "It/when/should" statement that describes what the test is supposed to do
- (Optional) list of prettier settings overrides (e.g. `captureWhitespace: "ignore"`, `printWidth: 1`, etc.)
- A block of code that doesn't include three consecutive new line characters. 

So something like this in `index.liquid`
```liquid
It should pretty print the contents of a capture tag with captureWhitespace ignore
printWidth: 1
captureWhitespace: "ignore"
{% capture _ %}
{%for element in col%}
{{element}}
{%endfor%}
{% endcapture %}  

It should print as is when strict
printWidth: 1
captureWhitespace: "strict"
{% capture _ %}
{%for element in col%}
{{element}}
{%endfor%}
{% endcapture %}
```

And this in `fixed.liquid`
```liquid
It should pretty print the contents of a capture tag with captureWhitespace ignore
printWidth: 1
captureWhitespace: "ignore"
{% capture _ %}
  {%for element in col%}
    {{ element }}
  {% endfor %}
{% endcapture %}  

It should print as is when strict
printWidth: 1
captureWhitespace: "strict"
{% capture _ %}
{%for element in col%}
{{element}}
{%endfor%}
{% endcapture %}
```

Reply:
Did my best! It already helped that I also created my own extension for Shopify, haha.

Makes sense! Thanks for the explanation. I'll fix it tomorrow!

Reply:
All test are passing now.

![image](https://github.com/Shopify/theme-tools/assets/47820758/fa37678a-d105-4500-9725-a0fd219b3234)